java/core/src/main/java/sleeper/core/partition/Partition.java

```diff
@@ -195,6 +195,9 @@ public Builder dimension(int dimension) {
         }
 
         public Partition build() {
+            if (childPartitionIds != null) {
+                this.leafPartition = childPartitionIds.isEmpty();
+            }
```

Comment:
This seems like a separate issue. I think there are some tests that rely on setting the flag to an incorrect value to test validation, but I guess they must not be valid otherwise they'd be failing now.

Probably best to remove this before we fix those tests.